readthedocs/search/api/v3/backend.py

```diff
@@ -0,0 +1,225 @@
+from functools import cached_property
+from itertools import islice
+
+from readthedocs.builds.models import Version
+from readthedocs.projects.models import Project
+from readthedocs.search.api.v3.queryparser import SearchQueryParser
+from readthedocs.search.faceted_search import PageSearch
+
+
+class Backend:
+
+    """
+    Parse the query, search, and return the projects used in the search.
+
+    :param arguments_required: If `True` and the user didn't provide
+     any arguments in the query, we don't perform the search.
+    :param default_all: If `True` and `arguments_required` is `False`
+     we search all projects by default, otherwise we search all projects
+     the user has access to.
+    :param max_projects: The maximum number of projects used in the search.
+     This limit is only applied for projects given explicitly,
+     not when we default to search all projects.
+    """
+
+    def __init__(
+        self,
+        *,
+        request,
+        query,
+        arguments_required=True,
+        default_all=False,
+        max_projects=100
```

Comment:
We can decrease or increase this number if we want, I just chose a number that isn't that big, an org with lots of projects may need a larger number? I'll check the numbers on .org and .com...